contrib/filter-repo-demos/git_lint_history.py

```diff
@@ -142,42 +151,47 @@ def lint_non_binary_blobs(blob, metadata):
     filename = '.git/info/tmpfile'
     with open(filename, "wb") as f:
       f.write(blob.data)
-    subprocess.check_call(lint_args.command + [filename])
+    subprocess.check_call(shlex.split(lint_args.command[0]) + [filename])
     with open(filename, "rb") as f:
       blob.data = f.read()
     os.remove(filename)
 
 def main(argv1=None):
-  global is_relevant, lint_args, tmpdir, cat_file_process, filter
+  global is_relevant, lint_args, tmpdir, cat_file_process, rfilter
 
   if isinstance(argv1, str):
     argv1 = shlex.split(argv1)
 
-  lint_args = parser.parse_args(argv1)
+  lint_args, rest = parser.parse_known_args(argv1)
   if not lint_args.command:
     raise SystemExit("Error: Need to specify a lint command")
-
+  if lint_args.relevant_glob:
+    if lint_args.relevant:
+      raise SystemExit("--relevant and --relevant-glob given both")
+    lint_args.relevant = 'return ' + ' or '.join("fnmatch.fnmatch(filename, %r)" % x.encode('utf-8') for x in lint_args.relevant_glob)
   if lint_args.filenames_important and not lint_args.relevant:
     lint_args.relevant = 'return True'
   if lint_args.relevant:
     body = lint_args.relevant
     exec('def is_relevant(filename):\n  '+'\n  '.join(body.splitlines()),
          globals())
     lint_args.filenames_important = True
-  args = fr.FilteringOptions.default_options()
-  args.force = True
+  args = fr.FilteringOptions.parse_args(rest, error_on_empty=False)
+  if args.partial:
+    # --refs ...  / --partial given, usual on non-clean clones
+    args.force = True
   if lint_args.filenames_important:
     tmpdir = tempfile.mkdtemp().encode()
     cat_file_process = subprocess.Popen(['git', 'cat-file', '--batch'],
                                         stdin = subprocess.PIPE,
                                         stdout = subprocess.PIPE)
-    filter = fr.RepoFilter(args, commit_callback=lint_with_real_filenames)
-    filter.run()
+    rfilter = fr.RepoFilter(args, commit_callback=lint_with_real_filenames)
+    rfilter.run()
```

Comment:
The rename from filter to rfilter appears to be gratuitous and unrelated.  Are you just trying to avoid a function defined elsewhere?  If so, that's probably a good change, but one that is not part of the functional changes being described and thus should really be part of its own commit.